vendors/espressif/boards/esp32/ports/ble/nimble/iot_ble_hal_common_gap.c

```diff
@@ -246,26 +246,40 @@ int prvGAPeventHandler( struct ble_gap_event * event,
                                              0 );
             }
 
-            if( ble_hs_cfg.sm_sc && ( event->pairing_req.authreq & BLE_SM_PAIR_AUTHREQ_SC ) &&
-                ( ble_sm_sc_ioa[ ble_hs_cfg.sm_io_cap ][ event->pairing_req.io_cap ] == BLE_SM_IOACT_NONE ) )
+            if( ble_hs_cfg.sm_sc && ( event->pairing_req.authreq & BLE_SM_PAIR_AUTHREQ_SC ))
             {
-                ESP_LOGE( TAG, "Just works in Secure Connections only mode" );
-
-                if( xBTCallbacks.pxPairingStateChangedCb != NULL )
+                if ( ble_sm_sc_ioa[ ble_hs_cfg.sm_io_cap ][ event->pairing_req.io_cap ] == BLE_SM_IOACT_NONE )
                 {
-                    xBTCallbacks.pxPairingStateChangedCb( eBTStatusFail, ( BTBdaddr_t * ) desc.peer_id_addr.val,
-                                                          eBTbondStateNone,
-                                                          eBTSecLevelNoSecurity,
-                                                          eBTauthFailInsuffSecurity );
+                    ESP_LOGE( TAG, "Just works in Secure Connections only mode" );
+
+                    if( xBTCallbacks.pxPairingStateChangedCb != NULL )
+                    {
+                        xBTCallbacks.pxPairingStateChangedCb( eBTStatusFail, ( BTBdaddr_t * ) desc.peer_id_addr.val,
+                                                              eBTbondStateNone,
+                                                              eBTSecLevelNoSecurity,
+                                                              eBTauthFailInsuffSecurity );
+                    }
+
+                    return BLE_SM_ERR_AUTHREQ;
+
                 }
+                else if (event->pairing_req.max_enc_key_size < BLE_SM_PAIR_KEY_SZ_MAX)
```

Comment:
The AFQP tests for pairing failed. After debugging the issue, I think max_enc_key_size parameter is not set within the event [here]( https://github.com/prasad-alatkar/amazon-freertos/blob/ble/NimBLE_fix_pairing_req_sc_only/vendors/espressif/esp-idf/components/nimble/nimble/nimble/host/src/ble_gap.c#L4504)

After adding the line `event.pairing_req.max_enc_key_size = req->max_enc_key_size;` , the test passes.
Can you fix the same ?